addon/services/sticky-manager.js

```diff
@@ -0,0 +1,245 @@
+      this.addEventListener(window, 'resize', this.handleResize);
+      this.manageStickyItems();
+    },
+
+    removeScrollListener() {
+      let container = this.get('container');
+      if (container) {
+        this.removeEventListener(container, 'scroll', this.handleScroll);
+        this.removeEventListener(window, 'resize', this.handleResize);
+      }
+    },
+
+    handleResize() {
+      this.throttleTask('manageStickyItems', 50);
+    },
+
+    handleScroll() {
+      this.throttleTask('manageStickyItems', 50);
+    },
+
+    manageStickyItems() {
+      let stickyItems = this.get('stickyItems');
+      if (stickyItems.length <= 0) {
+        return;
+      }
+
+      let container = this.get('container');
+      let scrollTop = scrollTopFor(container);
+      let containerTop = getRectForNode(container).top;
+
+      stickyItems.forEach((stickyItem) => {
+        let { handlePositioning } = stickyItem;
+
+        let { sticky, top, left, width } = this.evaluateStickyItem(
+          stickyItem,
+          scrollTop,
+          containerTop
+        );
+
+        this.updateStuckItems(stickyItem, sticky);
+
+        handlePositioning(sticky, top, left, width);
+      });
+
+      /*
+       * This call isn't in the original code, but there seems to be
+       * a difference between how the throttle implementations work
+       * between the React and Ember worlds, which meant that sticky
+       * items in Ember could end up in the wrong position after a
+       * scroll/resize event, until the next such event. This is a
+       * workaround that ensures that no sticky items will be left
+       * in the wrong positions more than 50ms after these events.
+       */
+      this.debounceTask('manageStickyItems', 50);
+    },
+
+    evaluateStickyItem(stickyItem, scrollTop, containerTop) {
+      let {
+        stickyNode,
+        placeHolderNode,
+        boundingElement,
+        offset,
+        disableWhenStacked,
+      } = stickyItem;
+
+      if (disableWhenStacked && stackedContent().matches) {
+        return {
+          sticky: false,
+          top: 0,
+          left: 0,
+          width: 'auto',
+        };
+      }
+
+      let stickyOffset = offset
+        ? this.getOffset(stickyNode) + parseInt(tokens.spacingLoose, 10)
+        : this.getOffset(stickyNode);
+
+      let scrollPosition = scrollTop + stickyOffset;
+      let placeHolderNodeCurrentTop =
+        getRectForNode(placeHolderNode).top - containerTop + scrollTop;
+      let top = containerTop + stickyOffset;
+      let width = getRectForNode(placeHolderNode).width;
+      let left = getRectForNode(placeHolderNode).left;
+
+      let sticky;
+
+      if (boundingElement == null) {
+        sticky = scrollPosition >= placeHolderNodeCurrentTop;
+      } else {
+        let stickyItemHeight = getRectForNode(stickyNode).height;
+        let stickyItemBottomPosition =
+          getRectForNode(boundingElement).bottom -
```

Comment:
should we still be using here `getBoundingClientRect` instead of `getRectForNode `....like React does? I don't think `getRectForNode` has `bottom` property, no?